wagtail/templatetags/wagtailcore_tags.py

```diff
@@ -98,12 +98,12 @@ def wagtail_release_notes_path():
 
 @register.simple_tag
 def wagtail_feature_release_whats_new_link():
-    return "https://guide.wagtail.org/en-latest/releases/new-in-wagtail-4-1/"
+    return "https://guide.wagtail.org/en-4.2.x/releases/new-in-wagtail-4-2/"
```

Comment:
We need to change this as well:

https://github.com/wagtail/wagtail/blob/667379f9ff9f202e242fb9ea935c48e276b60050/wagtail/admin/wagtail_hooks.py#L976-L977